src/Nethermind/Nethermind.Evm/EvmState.cs

```diff
@@ -162,32 +173,19 @@ internal EvmState(
             IsCreateOnPreExistingAccount = isCreateOnPreExistingAccount;
             if (stateForAccessLists is not null)
             {
-                // if we are sub-call, then we use the main collection for this transaction
-                _accessedAddresses = stateForAccessLists._accessedAddresses;
-                _accessedStorageCells = stateForAccessLists._accessedStorageCells;
-                _destroyList = stateForAccessLists._destroyList;
-                _createList = stateForAccessLists._createList;
-                _logs = stateForAccessLists._logs;
+                // if we are sub-call, then we use the existing access list for this transaction
+                _accessTracker = new(stateForAccessLists);
             }
             else
             {
-                // if we are top level, then we need to create the collections
-                _accessedAddresses = new JournalSet<Address>();
-                _accessedStorageCells = new JournalSet<StorageCell>();
-                _destroyList = new JournalSet<Address>();
-                _createList = new HashSet<AddressAsKey>();
-                _logs = new JournalCollection<LogEntry>();
+                // if we are top level, then we create a new access tracker
+                _accessTracker = new();
```

Comment:
Wonder if we could/should move this code upper?

Reply:
code has been simplified and refactored.